### ISSUE TITLE
Post Install and setup config session error message displayed after login

### ISSUE THREAD

**Issue by caugheygm** [opendocman/opendocman#313]
After a successful automated install of the community version of OpenDocMan I selected to login using admin and password credentials I had setup during the installation. At this point I was presented with a form (configuration params) which I filled in and saved. I then logged out and closed the browser window in preparation for starting a fresh OpenDocMan session. 

I then opened a new browser window and typed: http://localhost/opendocman and hit enter. I was immediately presented with a login screen and entered "admin" then "my_password" and hit enter. This created a clean window displaying nothing but the following error message: "**403 Access Forbidden by CSRFProtector!**" and the session ends here.

**To Reproduce**
Just attempt to login again and the same behavior will happen repeatedly. I've research the internet and tried to fix this issue myself. I've cleared browser cache, cookies, history and whatever else with no joy at this point.

**Expected behavior**
To launch OpenDocMan, logon with appropriate credentials and be presented with the users GUI and manage my documents.

**Screenshots**
Not applicable for this issue. 

**Desktop (Hewlett-Packard tower w/linux):**
 - OS: [Ubuntu 20.04.2 LTS]
 - Browser Firefox
 - Version:  [87.0 (64-bit)]

**Additional context**
I did a fresh OS installation about a month ago with clean supporting software packages, using Ubuntu package manager to do all the work. 

Everything is up-to-date including: php-7.4
perl 5, version 30, subversion 0 (v5.30.0) built for x86_64-linux-gnu-thread-multi
(with 50 registered patches, see perl -V for more detail) and etc.

**Comment by stephenlawrence** [comment]
That version of OpenDocMan are you using?

**Comment by caugheygm** [comment]
> What version of OpenDocMan are you using?

OpenDocMan v 1.4.4

**Comment by stephenlawrence** [comment]
@caugheygm I just performed a test installation on a hosting system and using a folder-based URL like you did I was able to install and then login without any issues.

The place where the base_url is computed is in functions.php line 881. 
Add this line after line 890:
`sprintf("%s://%s", $proto, $_SERVER['HTTP_HOST'].dirname($_SERVER['REQUEST_URI']);exit;`

Try to login again and let me know the output.

**Comment by caugheygm** [comment]
Good morning,
I inserted your line of code at line 891 of file 'functions.php' and tried it again. This experiment shows no error message but no page either just blank screen. I then reopened 'functions.php' and commented out your line of code to see what happens. The output shows the same error message as before with no other output.

**Comment by stephenlawrence** [comment]
Sorry, that line is missing echo:

`echo sprintf("%s://%s", $proto, $_SERVER['HTTP_HOST'].dirname($_SERVER['REQUEST_URI']);exit;`

**Comment by stephenlawrence** [comment]
@caugheygm If you would like, there is a branch (issue-313-csrf-magic) that switches out the CSRF library with a different one. The old one always seemed grumpy so wondering if the changes in this branch might resolve your issues.

**Comment by sarazavi** [comment]
> @caugheygm If you would like, there is a branch (issue-313-csrf-magic) that switches out the CSRF library with a different one. The old one always seemed grumpy so wondering if the changes in this branch might resolve your issues.

Hi
I have the same issue and resolved with your code.
but your code have a mistake and right code is:
echo sprintf("%s://%s", $proto, $_SERVER['HTTP_HOST'].dirname($_SERVER['REQUEST_URI']));exit;

**Comment by stephenlawrence** [comment]
> > @caugheygm If you would like, there is a branch (issue-313-csrf-magic) that switches out the CSRF library with a different one. The old one always seemed grumpy so wondering if the changes in this branch might resolve your issues.
> 
> Hi
> I have the same issue and resolved with your code.
> but your code have a mistake and right code is:
> echo sprintf("%s://%s", $proto, $_SERVER['HTTP_HOST'].dirname($_SERVER['REQUEST_URI']));exit;


@sarazavi  Yes, typo there, thanks. Did the issue-313-csrf-magic branch resolve your problem?

**Comment by sarazavi** [comment]
> > > @caugheygm If you would like, there is a branch (issue-313-csrf-magic) that switches out the CSRF library with a different one. The old one always seemed grumpy so wondering if the changes in this branch might resolve your issues.
> > 
> > 
> > Hi
> > I have the same issue and resolved with your code.
> > but your code have a mistake and right code is:
> > echo sprintf("%s://%s", $proto, $_SERVER['HTTP_HOST'].dirname($_SERVER['REQUEST_URI']));exit;
> 
> @sarazavi Yes, typo there, thanks. Did the issue-313-csrf-magic branch resolve your problem?

Yes. This line resolved my problem.

**Comment by caugheygm** [comment]
Hi Stephen, sorry about just getting back to you this slow, I have an update regarding my issue. After about a dozen installs with OpenDocMan-1.4.4 and having the same error message "403 Access Forbidden by CSRFProtector!" every time it's now working just fine. I have no explanation for this. I did another fresh install in preparation for adding your revised code string and trying again. However, this last time I did the auto-installation, did the initial admin login and edited the configuration file and then logged off. Same process I'd done all the other attempts.

Before editing 'functions.php' I decided to try and logon again and then like magic OpenDocMan worked and has kept on working without any issues. Go figure... 
If anything goes wrong down the road I'll edit that code string in and drop you a note as to the results. Maybe I should just edit it in right now?
Thanks, Greg
